handlers.go

```diff
@@ -81,3 +97,51 @@ func (dh *DeviceHandlers) writeObject(w http.ResponseWriter, device *Device) {
 		return
 	}
 }
+
+func pageAndLimitWrapper(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var err error
+		var limit, page int
+
+		limitStr := r.URL.Query().Get("limit")
```

Comment:
The whole 
```
limitStr := r.URL.Query().Get("limit")
		if limitStr == "" {
			limit = 100
		} else {
			limit, err = convertToPositiveInteger(limitStr)
			if err != nil {
				fmt.Println(err.Error())
				w.WriteHeader(http.StatusBadRequest)
				return
			}
		}
```

Could is called here two times - we could make a func like `func readIntFromQueryParameter(url *url.URl, param string, defaultValue int) (int, error)`